### ISSUE TITLE
MathML is unsupported

### ISSUE THREAD

**Issue by lukewarlow** [angular/angular#55608]
### Which @angular/* package(s) are the source of the bug?

core

### Is this a regression?

No

### Description

When trying to use [MathML](https://developer.mozilla.org/en-US/docs/Web/MathML) in an angular template you get a warning that `<math>` it's an unrecognised element. If you disable this warning by adding `schemas: [NO_ERRORS_SCHEMA]` it doesn't render correctly. At first glance they seem correct because the produced elements are in the MathML namespace (something other frameworks can get wrong) but the element's prototype is just `Element` rather than `MathMLElement`.

### Please provide a link to a minimal reproduction of the bug

https://stackblitz.com/edit/github-tdmxzz?file=src%2Fmain.ts

### Please provide the exception or error you saw

```true
Error in src/main.ts (12:3)
':math:math' is not a known element:
1. If ':math:math' is an Angular component, then verify that it is included in the '@Component.imports' of this component.
2. To allow any element add 'NO_ERRORS_SCHEMA' to the '@Component.schemas' of this component.
```
```


### Please provide the environment you discovered this bug in (run `ng version`)

_No response_

### Anything else?

The template I tried is below. 

```html
<math>
  <mfrac>
    <mn>1</mn>
    <msqrt>
      <mn>2</mn>
    </msqrt>
  </mfrac>
</math>
```

**Comment by JoostK** [comment]
Intersting, this should work as we do parse this as `math` namespace, but then fail to validate it correctly.

**Comment by lukewarlow** [comment]
> Intersting, this should work as we do parse this as math namespace, but then fail to validate it correctly.

Yeah, there's definitely some level of support for it, the namespace appears correct but then it seems to get the prototype wrong (not even sure how that's possible?). As a guess it's not using createElementNS(), it's doing something else.

**Comment by JeanMeche** [comment]
Yeah the [compiler output](https://jeanmeche.github.io/angular-compiler-output/?template=PG1hdGg%2BCiAgxIBmcmFjxIXEh8SHxIBuPjE8L23EksSGxI%2FEgHNxcnTEjsSPxJDElz4yxJXEo8SZxJDElsScxJ7EoMSmxIrEjMSFxKbEgsSECg%3D%3D) clearly has the MathML support in it. 

The `DomRenderer` relies on `createElementNS()` but it looks like the namespace is wrong.

**Comment by JeanMeche** [comment]
With #55622 we'll at least get the support but it will still require `NO_ERRORS_SCHEMA `.

**Comment by JoostK** [comment]
> With #55622 we'll at least get the support for MathML but it will still require `NO_ERRORS_SCHEMA `. 

Yeah that error is because DomSchemaChecker is looking up the definition using the prefixed tagname `:math:math` which fails.

**Comment by lukewarlow** [comment]
@JeanMeche great catch! I was just helping someone else with a similar issue where createElementNS was returning an Element and it was an invalid namespace which brought me back here, you're one step ahead though

**Comment by lukewarlow** [comment]
Would be good to reopen this issue until https://github.com/angular/angular/pull/55622 is merged (or go ahead and merge that as it's approved)

**Comment by JeanMeche** [comment]
@lukewarlow Yes, we're looking at landing this "fix" into 17.3 and the upcomming 18.0 !

**Comment by lukewarlow** [comment]
Closing this now that the second PR has been merged.
